packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -88,9 +88,13 @@ const propTypes = {
    */
   onChangeRaw: PropTypes.func,
   /**
-   * A callback function to execute when clicking outside of the picker to dismiss it.
+   * **Deprecated**, A callback function to execute when clicking outside of the picker to dismiss it. Resolves to `onDismiss`.
    */
   onClickOutside: PropTypes.func,
+  /**
+   * A callback function to execute when clicking outside of the picker to dismiss it.
+   */
+  onDismiss: PropTypes.func,
```

Comment:
Do we still want to name this `onDismiss` or name is `onRequestClose`, similar to terra-popup and other places?

Reply:
I think we can name it `onRequestClose`. And handle the prop [over here](https://github.com/cerner/terra-framework/blob/main/packages/terra-date-picker/src/react-datepicker/index.jsx#L425), rather than handling it in setOpen.

Reply:
[onRequestClose](https://github.com/cerner/terra-framework/blob/main/packages/terra-date-picker/src/react-datepicker/index.jsx#L425) is only called for the popup picker in larger breakpoints. It doesn't get called for the portal picker in smaller breakpoints os setOpen would handle both workflows. Perhaps should we renamed `onDismiss` to `onRequestClose` to be more consistent with terra-popup and other places.

Reply:
Updated here https://github.com/cerner/terra-framework/pull/1224/commits/2f250c5b8b3b3f266e82368a847a018d45853c41